manifests/params.pp

```diff
@@ -76,6 +76,10 @@
   # Custom puppetmaster
   $puppetmaster              = $::puppetmaster
 
+  # Hashes containing additional settings
+  $additional_settings       = {}
```

Comment:
I think right now consistency with the others (so leave main) out is better. IIRC anything in main also counts for [master] and [agent] so that means it's really the generic additional settings. So leave it as it is now, but I reserve the right to change my opinion again ;)